src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            ensureCapacity();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != size) {
+            rangeCheck(index);
+        }
+        if (size == elementData.length) {
+            ensureCapacity();
+        }
+        if (index == size) {
+            elementData[size] = value;
+        } else {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > elementData.length) {
+            ensureCapacity();
+        }
+        int indexOfList = 0;
+        int forLoop = size;
+        for (int i = size; i < forLoop + list.size(); i++) {
+            elementData[i] = list.get(indexOfList);
+            size++;
+            indexOfList++;
+        }
```

Comment:
example of implementation


```suggestion
        for (int i = 0; i < list.size(); i++) {
            resize();
            elements[size] = list.get(i);
            size++;
        }
```